core/src/main/java/com/sequenceiq/cloudbreak/rotation/saltboot/UserDataRotationExecutor.java

```diff
@@ -0,0 +1,209 @@
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
+import com.sequenceiq.cloudbreak.converter.spi.ResourceToCloudResourceConverter;
+import com.sequenceiq.cloudbreak.converter.spi.StackToCloudStackConverter;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationExecutor;
+import com.sequenceiq.cloudbreak.rotation.secret.SecretRotationException;
+import com.sequenceiq.cloudbreak.rotation.secret.step.SecretRotationStep;
+import com.sequenceiq.cloudbreak.rotation.secret.userdata.UserDataRotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.userdata.UserDataSecretModifier;
+import com.sequenceiq.cloudbreak.service.image.userdata.UserDataService;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+import com.sequenceiq.cloudbreak.service.secret.domain.RotationSecret;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.cloudbreak.util.UserDataReplacer;
+import com.sequenceiq.common.api.type.CommonStatus;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Component
+public class UserDataRotationExecutor implements RotationExecutor<UserDataRotationContext> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserDataRotationExecutor.class);
+
+    @Inject
+    private StackDtoService stackService;
+
+    @Inject
+    private SecretService secretService;
+
+    @Inject
+    private UserDataService userDataService;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private ResourceToCloudResourceConverter resourceToCloudResourceConverter;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Inject
+    private StackToCloudStackConverter cloudStackConverter;
+
+    @Inject
+    private Optional<List<UserDataSecretModifier>> userDataModifiers;
+
+    private final Map<Class<? extends UserDataSecretModifier>, UserDataSecretModifier> userDataModifierMap = new ConcurrentHashMap<>();
+
+    @PostConstruct
+    public void setUp() {
+        userDataModifiers.ifPresent(modifiers -> modifiers.forEach(modifier -> userDataModifierMap.put(modifier.getClass(), modifier)));
+    }
+
+    @Override
+    public void rotate(UserDataRotationContext rotationContext) {
+        LOGGER.info("Userdata rotation is requested {}", rotationContext);
+        modifyUserData(rotationContext, useNewSecret());
+        updateLaunchTemplateIfNeeded(rotationContext.getResourceCrn());
```

Comment:
same implementation in rotate and rollback, pls move it into method, only the function differs